src/utils/safenet.ts

```diff
@@ -0,0 +1,49 @@
+import type { SafeNetBalanceEntity, SafeNetConfigEntity } from '@/store/safenet'
+import { TokenType, type SafeBalanceResponse } from '@safe-global/safe-gateway-typescript-sdk'
+
+const enum SafenetChainType {
+  SOURCE = 'source',
+  DESTINATION = 'destination',
+}
+
+const isSupportedChain = (chainId: number, safeNetConfig: SafeNetConfigEntity, chainType: SafenetChainType) => {
+  return chainType === SafenetChainType.SOURCE
+    ? safeNetConfig.chains.sources.includes(chainId)
+    : safeNetConfig.chains.destinations.includes(chainId)
+}
+
+const convertSafeNetBalanceToSafeClientGatewayBalance = (
+  safeNetBalance: SafeNetBalanceEntity,
+  safeNetConfig: SafeNetConfigEntity,
+  chainId: number,
+): SafeBalanceResponse => {
+  const balances: SafeBalanceResponse = {
+    fiatTotal: safeNetBalance.fiatTotal,
+    items: [],
+  }
+
+  for (const [tokenName, balance] of Object.entries(safeNetBalance)) {
+    const tokenAddress = safeNetConfig.tokens[tokenName][chainId]
+    if (!tokenAddress) {
+      continue
+    }
+
+    balances.items.push({
+      tokenInfo: {
+        type: TokenType.ERC20,
+        address: tokenAddress,
+        decimals: 18,
+        symbol: tokenName,
+        name: `${tokenName} (SafeNet)`,
+        logoUri: '',
+      },
+      balance,
+      fiatBalance: '0',
+      fiatConversion: '0',
+    })
+  }
+
+  return balances
```

Comment:
nitpick: good candidate for a reducer? or too cryptic? 🙈 

**untested**
```js
return Object.entries(safeNetBalance).reduce((agg, [tokenName, balance] ) => ({
  ...agg,
  items: [
    ...agg.items, 
    ...(safeNetConfig.tokens[tokenName][chainId] ? [{
      tokenInfo: {
        type: TokenType.ERC20,
        address: tokenAddress,
        decimals: 18,
        symbol: tokenName,
        name: `${tokenName} (SafeNet)`,
        logoUri: '',
      },
      balance,
      fiatBalance: '0',
      fiatConversion: '0',
    }]: []) ]
}), {
  fiatTotal: safeNetBalance.fiatTotal,
  items: [],
})
```

Reply:
I personally find that many spread operators paired with a ternary rather cryptic

Reply:
That's fair